macros/sql/dedupe.sql

```diff
@@ -0,0 +1,41 @@
+{%- macro dedupe(table, group_by, order_by=none) -%}
+    {{ return(adapter.dispatch('dedupe', 'dbt_utils')(table, group_by, order_by=order_by)) }}
+{% endmacro %}
+
+{%- macro default__dedupe(table, group_by, order_by=none) -%}
+
+    select
+        {{ dbt_utils.star(table, relation_alias='deduped') | indent }}
+    from (
+        select
+            _inner.*,
+            row_number() over (
+                partition by {{ group_by }}
+                {% if order_by != none -%}
+                order by {{ order_by }}
+                {%- endif %}
+            ) as rn
+        from {{ table }} as _inner
+    ) as deduped
```

Comment:
Perhaps the only place in all of dbt-dom that a subquery is preferable over a CTE 😮